src/components/Agents/components/NewsletterModal/components/Input.tsx

```diff
@@ -0,0 +1,209 @@
+import { Text } from './Text';
+import { cn } from '@utils/cn';
+import { cva, type VariantProps } from 'class-variance-authority';
+import { forwardRef, type PropsWithChildren } from 'react';
+
+const inputVariants = cva(
+  'w-full font-plex-sans border border-transparent overflow-hidden bg-transparent flex items-center text-gray-dark-12',
+  {
+    variants: {
+      size: {
+        xs: 'min-h-[24px] px-8 text-xs gap-6 rounded-4',
+        sm: 'min-h-[32px] px-8 text-sm gap-6 rounded-8',
+        md: 'min-h-[40px] px-3 gap-6 rounded-8',
+        lg: 'min-h-[48px] text-[1.8rem] px-4 gap-12 rounded-12',
+      },
+      variant: {
+        outline:
+          'border-gray-dark-7 focus-within:outline outline-1 outline-transparent outline-offset-0 focus-within:outline-gray-dark-8 focus-within:border-gray-dark-8',
+        ghost: 'px-0 rounded-none',
+      },
+      status: {
+        isLoading: 'border-gray-dark-4 animate-pulse bg-gray-dark-4',
+        error:
+          'border-red-dark-8 focus-within:border-red-dark-8 focus-within:outline-red-dark-8',
+        disabled:
+          'cursor-not-allowed bg-gray-dark-3 border-gray-dark-7 text-gray-dark-11',
+      },
+    },
+    compoundVariants: [
+      {
+        variant: 'ghost',
+        status: 'isLoading',
+        className:
+          'border border-gray-dark-4 animate-pulse bg-gray-dark-4 rounded-lg',
+      },
+      {
+        variant: 'ghost',
+        status: 'error',
+        className:
+          'border-0 rounded-none border-b border-red-dark-8 focus-within:border-red-dark-8',
+      },
+      {
+        variant: 'ghost',
+        status: 'disabled',
+        className: 'bg-transparent border-none cursor-not-allowed',
+      },
+    ],
+    defaultVariants: {
+      variant: 'outline',
+      size: 'sm',
+    },
+  },
+);
+
+export type InputVariants = VariantProps<typeof inputVariants>;
+
+export type InputStatus =
+  | 'validating'
+  | 'invalid'
+  | 'valid'
+  | 'pending'
+  | 'debouncing'
+  | 'other';
+
+export type InputRootProps = React.HTMLAttributes<HTMLDivElement> &
+  InputVariants &
+  PropsWithChildren & {
+    isLoading?: boolean;
+    error?: boolean;
+    disabled?: boolean;
+    hidden?: boolean;
+  };
+
+const Root = forwardRef<HTMLDivElement, InputRootProps>(
+  (
+    {
+      variant,
+      size,
+      isLoading,
+      children,
+      className,
+      error,
+      hidden,
+      disabled,
+      ...props
+    },
+    ref,
+  ) => {
+    const status = isLoading
+      ? 'isLoading'
+      : disabled
+        ? 'disabled'
+        : error
+          ? 'error'
+          : undefined;
```

Comment:
Nested ternary's can be hard to read and maintain. Maybe we can make it a bit easier by:

```suggestion
const status = 
  (isLoading && "isLoading") ??
  (disabled && "disabled") ??
  (error && "error");
```

Reply:
Ok, thank you. Just a note, all files from `Agents/components/NewsletterModal/components` are copy pasted as is from `agents-ui`.